cognite/pygen/_query/executer.py

```diff
@@ -0,0 +1,277 @@
+import datetime
+import time
+import warnings
+from collections.abc import Iterator, Sequence
+from dataclasses import dataclass, field
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes import data_modeling as dm
+from cognite.client.data_classes.aggregations import Count
+from cognite.client.exceptions import CogniteAPIError
+
+from cognite.pygen._query.constants import (
+    ACTUAL_INSTANCE_QUERY_LIMIT,
+    IN_FILTER_CHUNK_SIZE,
+    INSTANCE_QUERY_LIMIT,
+    MINIMUM_ESTIMATED_SECONDS_BEFORE_PRINT_PROGRESS,
+    PRINT_PROGRESS_PER_N_NODES,
+    SEARCH_LIMIT,
+)
+from cognite.pygen._query.processing import QueryResultCleaner
+from cognite.pygen._query.step import QueryBuildStep, QueryResultStep
+
+
+class QueryReducingBatchSize(UserWarning):
+    """Raised when a query is too large and the batch size must be reduced."""
+
+    ...
+
+
+@dataclass
+class Progress:
```

Comment:
`Progress` class here seems to be doing a bit more than a "normal" dataclass. Might be personal preference of mine, but I usually see dataclasses as doing less as possible.

Reply:
I use `dataclass` to quickly instantiate the state. All the progress bar is doing is logging progress, and thus needs to maintain state.